src/internal/Handler.ts

```diff
@@ -44,6 +45,8 @@ export class Handler<E, A> implements Fx<E, A>, Pipeable {
               ir = i.next(hr)
             } else if (Fork.is(ir.value)) {
               ir = i.next(yield new Fork({ ...ir.value.arg, context: [...ir.value.arg.context, this] }) as any)
+            } else if (GetHandlerContext.is(ir.value)) {
+              ir = i.next([this, ...(yield ir.value) as any])
```

Comment:
I don't love that Handler now knows about 2 specific effects, Fork and GetHandlerContext.  I think it will be possible to remove knowledge of Fork, and just rely on GetHandlerContext, but it'll take a bit of work to get there.

Probably something for a followup PR.